src/Sharpliner.Model/GHActions/Permissions.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+
+namespace Sharpliner.Model.GHActions
+{
+    public enum GitHubPermissionScope
+    {
+        Actions,
+        Checks,
+        Contents,
+        Deployments,
+        Issues,
+        Packages,
+        PullRequests,
+        RepositoryProjects,
+        SecurityEvents,
+        Statuses,
+    }
+
+    public enum GitHubPermission
+    {
+        None,
+        Read,
+        Write
+    }
+
+    public record Permissions
+    {
+        public HashSet<GitHubPermissionScope> Read { get; } = new();
+        public HashSet<GitHubPermissionScope> Write { get; } = new();
```

Comment:
I'm going to have some fun serialising this, but the API is so much nicer.

Reply:
Serializing is a piece of cake actually. Look up `IYamlConvertible` in our code and see how it is implemented.